wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -280,7 +280,10 @@ private[wallet] trait UtxoHandling extends WalletLogger {
 
     for {
       utxo <- utxoF
-      written <- spendingInfoDAO.create(utxo)
+      written <- spendingInfoDAO.createUnless(utxo) {
+        (foundUtxo, utxoToCreate) =>
+          (foundUtxo.state == BroadcastSpent || foundUtxo.state == Reserved) && (utxoToCreate.state == PendingConfirmationsReceived || utxoToCreate.state == ConfirmedReceived)
```

Comment:
I think this can be more general

```scala
          (foundUtxo.state.isInstanceOf[SpentState]) && (utxoToCreate.state
            .isInstanceOf[ReceivedState])
```